### ISSUE TITLE
Can't populate the RadAutoCompleteTextView with a value programmatically 

### ISSUE THREAD

**Issue by tsonevn** [telerik/nativescript-ui-feedback#385]
_From @vbudilov on November 5, 2017 3:0_

I'm working on functionality that will allow users to click on an icon (any image) and that will populate the RadAutoCompleteTextView with a default value. The UI also allows the user to enter the value in himself, that's why I'm using the AutoComplete component. For some reason, clicking the icon kicks off all of the necessary events, but I don't see the text showing up in the AutoComplete text field.  
![image](https://user-images.githubusercontent.com/2776052/32411527-68c85750-c1b3-11e7-899f-cd5a9619a9ee.png)

The code that's kicked off when the icon is tapped:
```
    public onTokenAdded(args) {
        console.log("Token added: " + args);
    }

    public onTokenRemoved(args) {
        console.log("Token removed: " + args);

    }

    selectIcon(icon:ActivityIcon) {
        console.log("Selected " + icon.displayName);
        this.autocmp.autoCompleteTextView.removeAllTokens();
        this.autocmp.autoCompleteTextView.addToken(new TokenModel(icon.displayName, "res://" + icon.localUrl));
    }
```
The UI: 
```
        <RadAutoCompleteTextView #autocmp [items]="suggestionTokens"
                                 maximumTokensCount="1"
                                 suggestMode="Suggest" displayMode="Plain"
                                 completionMode="Contains"
                                 (tokenAdded)="onTokenAdded($event)"
                                 (tokenRemoved)="onTokenRemoved($event)">
            <SuggestionView tkAutoCompleteSuggestionView suggestionViewHeight="120">
                <ng-template tkSuggestionItemTemplate let-item="item">
                    <StackLayout orientation="vertical" padding="10">
                        <Label [text]="item.text"></Label>
                    </StackLayout>
                </ng-template>
            </SuggestionView>
        </RadAutoCompleteTextView>
        <Label marginTop="10" text="Suggestions:" fontSize="12"></Label>
        <RadListView [items]="icons"  >
            <ng-template tkListItemTemplate let-item="item">
                <StackLayout  orientation="vertical" class="icon-with-text" (tap)="selectIcon(item)">
                    <Image  src="res://{{item.localUrl}}" stretch="none" ></Image>
                    <Label class="icon-text" textWrap="true" [text]="item.displayName"> </Label>
                </StackLayout>
            </ng-template>
            <ListViewLinearLayout tkListViewLayout scrollDirection="Horizontal"></ListViewLinearLayout>
        </RadListView>
```

Clicking on the icons produces the needed events:
```
JS: Selected football
JS: Token added: [object Object]
JS: Selected soccer
JS: Token removed: [object Object]
JS: Token added: [object Object]
```

Is this functionality actually supported? I couldn't find any docs/issues discussing this use case in enough detail. 

Thanks,
Vladimir

_Copied from original issue: NativeScript/NativeScript#5036_

**Comment by tsonevn** [comment]
Hi @vbudilov,
Thank you for reporting this issue.
I tested this case on my side and indeed there is a real issue and the events for the AutoCompleteTextView will not be fired while using Images in the Suggestion's template. 
This problem is reproducible also in a not-Angular project. Example:
```
<navigation:ExamplePage loaded="onPageLoaded" xmlns:navigation="navigation/example-page" xmlns:au="nativescript-pro-ui/autocomplete" xmlns="http://www.nativescript.org/tns.xsd">
    <StackLayout ios:backgroundColor="#CDCECE" padding="5">
        <Label text="Select country:"/>
        <!-- >> autocomplete-events-xml -->
        <au:RadAutoCompleteTextView id="autocmp" items="{{ dataItems }}"
          suggestMode="Suggest" displayMode="Tokens" 
          suggestionViewBecameVisible="{{ onSuggestionViewBecameVisible }}"
          didAutoComplete="{{ onDidAutoComplete }}"
          tokenAdded="{{ onTokenAdded}}" tokenRemoved="{{ onTokenRemoved }}"
          tokenSelected= "{{ onTokenSelected }}" tokenDeselected= "{{ onTokenDeselected }}">
            <au:RadAutoCompleteTextView.suggestionView>
                <au:SuggestionView suggestionViewHeight="300">
                    <au:SuggestionView.suggestionItemTemplate>
                        <StackLayout orientation="vertical" padding="10">
                        <Image  src="res://icon" stretch="none" />
                            <Label text="{{ text }}"></Label>
                        </StackLayout>
                    </au:SuggestionView.suggestionItemTemplate>
                </au:SuggestionView>
            </au:RadAutoCompleteTextView.suggestionView>
        </au:RadAutoCompleteTextView>
        <!-- << autocomplete-events-xml -->
        <StackLayout orientation="vertical" marginTop="5">
            <Label marginBottom="12" text="{{ eventsText }}"/>
            <Label text="{{ eventName1 }}"/>
            <Label text="{{ eventName2 }}"/>
            <Label text="{{ eventName3 }}"/>
            <Label text="{{ eventName4 }}"/>
            <Label text="{{ eventName5 }}"/>
        </StackLayout>
    </StackLayout>
</navigation:ExamplePage>
```

For further info, please keep track on the issue.

**Comment by venkateshsj** [comment]
Would love to see this feature implemented soon...!👍

**Comment by zbranzov** [comment]
A fix is released in nativescript-ui-autocomplete@3.9.0. Enjoy!
